JsParserException compiling ziplineJs.kt

Our CI system caught a crash in the Kotlin compiler:

```
> Task :myproject:compileTestDevelopmentExecutableKotlinJs FAILED
e: org.jetbrains.kotlin.backend.common.CompilationException: Back-end: Please report this problem https://kotl.in/issue
/Volumes/Development/zipline/zipline/src/jsMain/kotlin/app/cash/zipline/ziplineJs.kt:36:1
Problem with `class Zipline {
  internal constructor(userSerializersModule: SerializersModule) /* primary */ {
    super/*Any*/()
    /* <init>() */

  }

  private val eventListener: <no name provided>
    field = { // BLOCK
      local class <no name provided> : EventListener {
        constructor() /* primary */ {
          super/*EventListener*/()
          /* <init>() */

        }

        override fun serviceLeaked(zipline: Zipline, name: String) {
          <this>.<get-eventListenerService>().serviceLeaked(name = name)
        }

        /* fake */ override fun bindService(zipline: Zipline, name: String, service: ZiplineService)
        /* fake */ override fun takeService(zipline: Zipline, name: String, service: ZiplineService)
        /* fake */ override fun callStart(zipline: Zipline, call: Call): Any?
        /* fake */ override fun callEnd(zipline: Zipline, call: Call, result: CallResult, startValue: Any?)
        /* fake */ override fun applicationLoadStart(applicationName: String, manifestUrl: String?): Any?
        /* fake */ override fun applicationLoadSkipped(applicationName: String, manifestUrl: String, startValue: Any?)
        /* fake */ override fun applicationLoadSuccess(applicationName: String, manifestUrl: String?, zipline: Zipline, startValue: Any?)
        /* fake */ override fun applicationLoadFailed(applicationName: String, manifestUrl: String?, exception: Exception, startValue: Any?)
        /* fake */ override fun downloadStart(applicationName: String, url: String): Any?
        /* fake */ override fun downloadEnd(applicationName: String, url: String, startValue: Any?)
        /* fake */ override fun downloadFailed(applicationName: String, url: String, exception: Exception, startValue: Any?)
        /* fake */ override fun manifestParseFailed(applicationName: String, url: String?, exception: Exception)
        /* fake */ override fun moduleLoadStart(zipline: Zipline, moduleId: String): Any?
        /* fake */ override fun moduleLoadEnd(zipline: Zipline, moduleId: String, startValue: Any?)
        /* fake */ override fun initializerStart(zipline: Zipline, applicationName: String): Any?
        /* fake */ override fun initializerEnd(zipline: Zipline, applicationName: String, startValue: Any?)
        /* fake */ override fun mainFunctionStart(zipline: Zipline, applicationName: String): Any?
        /* fake */ override fun mainFunctionEnd(zipline: Zipline, applicationName: String, startValue: Any?)
        /* fake */ override fun ziplineCreated(zipline: Zipline)
        /* fake */ override fun ziplineClosed(zipline: Zipline)
        /* fake */ override operator fun equals(other: Any?): Boolean
        /* fake */ override fun hashCode(): Int
        /* fake */ override fun toString(): String
      }

      <no name provided>()
    }
    private get

  @OptIn(markerClass = [DelicateCoroutinesApi::class])
  internal val endpoint: Endpoint
    field = Endpoint(scope = GlobalScope, userSerializersModule = userSerializersModule, eventListener = EventListenerAdapter(delegate = <this>.<get-eventListener>(), zipline = <this>), outboundChannel = { // BLOCK
      local class <no name provided> : CallChannel {
        constructor() /* primary */ {
          super/*Any*/()
          /* <init>() */

        }

        @Suppress(names = ["UnsafeCastFromDynamic"])
        private val jsOutboundChannel: CallChannel
          private get(): CallChannel {
            return js(code = "globalThis." + <get-outboundChannelName>()) /*~> CallChannel */
          }

        override fun serviceNamesArray(): Array<String> {
          return <this>.<get-jsOutboundChannel>().serviceNamesArray()
        }

        override fun call(callJson: String): String {
          return <this>.<get-jsOutboundChannel>().call(callJson = callJson)
        }

        override fun disconnect(instanceName: String): Boolean {
          return <this>.<get-jsOutboundChannel>().disconnect(instanceName = instanceName)
        }

        /* fake */ override operator fun equals(other: Any?): Boolean
        /* fake */ override fun hashCode(): Int
        /* fake */ override fun toString(): String
      }

      <no name provided>()
    })
    internal get

  val json: Json
    field = <this>.<get-endpoint>().<get-json>()
    get

  internal val serviceNames: Set<String>
    internal get(): Set<String> {
      return <this>.<get-endpoint>().<get-serviceNames>()
    }

  internal val clientNames: Set<String>
    internal get(): Set<String> {
      return <this>.<get-endpoint>().<get-clientNames>()
    }

  internal val eventLoop: EventLoop
    field = <this>.<get-endpoint>().take<EventLoop>(name = <get-eventLoopName>(), adapter = Adapter(serializers = listOf<KSerializer<*>>(elements = [])))
    internal get

  internal val eventListenerService: EventListenerService
    field = <this>.<get-endpoint>().take<EventListenerService>(name = <get-eventListenerName>(), adapter = Adapter(serializers = listOf<KSerializer<*>>(elements = [])))
    internal get

  internal val console: Console
    field = <this>.<get-endpoint>().take<Console>(name = <get-consoleName>(), adapter = Adapter(serializers = listOf<KSerializer<*>>(elements = [])))
    internal get

  fun <T : ZiplineService> bind(name: String, instance: T) {
    error(message = "unexpected call to Zipline.bind: is the Zipline plugin configured?")
  }

  @PublishedApi
  internal fun <T : ZiplineService> bind(name: String, service: T, adapter: ZiplineServiceAdapter<T>) {
    <this>.<get-endpoint>().bind<T>(name = name, service = service, adapter = adapter)
  }

  fun <T : ZiplineService> take(name: String, scope: ZiplineScope = ZiplineScope()): T {
    error(message = "unexpected call to Zipline.take: is the Zipline plugin configured?")
  }

  @PublishedApi
  internal fun <T : ZiplineService> take(name: String, scope: ZiplineScope = ZiplineScope(), adapter: ZiplineServiceAdapter<T>): T {
    return <this>.<get-endpoint>().take<T>(name = name, scope = scope, adapter = adapter)
  }

  companion object Companion {
    private constructor() /* primary */ {
      super/*Any*/()
      /* <init>() */

    }

    fun get(serializersModule: SerializersModule = EmptySerializersModule()): Zipline {
      val theOnlyZipline: Zipline? = <get-THE_ONLY_ZIPLINE>()
      when {
        EQEQ(arg0 = theOnlyZipline, arg1 = null).not() -> { // BLOCK
          require(value = EQEQ(arg0 = serializersModule, arg1 = theOnlyZipline.<get-endpoint>().<get-userSerializersModule>()), lazyMessage = local fun <anonymous>(): Any {
            return "get() called multiple times with non-equal serializersModule instances"
          }
)
          return theOnlyZipline
        }
      }
      return Zipline(userSerializersModule = serializersModule).apply<Zipline>(block = local fun Zipline.<anonymous>() {
        $this$apply.bind<JsPlatform>(name = <get-jsPlatformName>(), service = GlobalBridge, adapter = Adapter(serializers = listOf<KSerializer<*>>(elements = [])))
        <set-THE_ONLY_ZIPLINE>(<set-?> = $this$apply)
      }
)
    }

    /* fake */ override operator fun equals(other: Any?): Boolean
    /* fake */ override fun hashCode(): Int
    /* fake */ override fun toString(): String
  }

  /* fake */ override operator fun equals(other: Any?): Boolean
  /* fake */ override fun hashCode(): Int
  /* fake */ override fun toString(): String
}
```

Here’s the stacktrace:

```
Details: Internal error in body lowering: org.jetbrains.kotlin.com.google.gwt.dev.js.parserExceptions.JsParserException: missing name after . operator at (51, 19)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.ThrowExceptionOnErrorReporter.error(ThrowExceptionReporter.kt:26)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.ThrowExceptionOnErrorReporter.error(ThrowExceptionReporter.kt:22)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Context.reportError(Context.java:342)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.TokenStream.reportSyntaxError(TokenStream.java:1441)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.reportError(Parser.java:80)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.mustMatchToken(Parser.java:73)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.memberExprTail(Parser.java:1024)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.memberExpr(Parser.java:1011)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.unaryExpr(Parser.java:928)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.mulExpr(Parser.java:881)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.addExpr(Parser.java:867)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.shiftExpr(Parser.java:857)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.relExpr(Parser.java:840)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.eqExpr(Parser.java:830)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.bitAndExpr(Parser.java:821)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.bitXorExpr(Parser.java:812)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.bitOrExpr(Parser.java:803)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.andExpr(Parser.java:793)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.orExpr(Parser.java:783)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.condExpr(Parser.java:769)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.assignExpr(Parser.java:752)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.expr(Parser.java:738)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.statementHelper(Parser.java:650)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.statement(Parser.java:339)
	at org.jetbrains.kotlin.com.google.gwt.dev.js.rhino.Parser.parse(Parser.java:131)
	at org.jetbrains.kotlin.js.parser.ParserUtilsKt.parseExpressionOrStatement(parserUtils.kt:62)
	at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.JsCodeKt.translateJsCodeIntoStatementList(jsCode.kt:63)
	at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.JsCodeKt.translateJsCodeIntoStatementList(jsCode.kt:28)
	at org.jetbrains.kotlin.ir.backend.js.lower.JsCodeOutlineTransformer.outlineJsCodeIfNeeded(JsCodeOutliningLowering.kt:137)
	at org.jetbrains.kotlin.ir.backend.js.lower.JsCodeOutlineTransformer.visitCall(JsCodeOutliningLowering.kt:129)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitCall(IrElementTransformerVoid.kt:215)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitCall(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.expressions.IrCall.accept(IrCall.kt:25)
	at org.jetbrains.kotlin.ir.expressions.IrExpression.transform(IrExpression.kt:28)
	at org.jetbrains.kotlin.ir.expressions.IrTypeOperatorCall.transformChildren(IrTypeOperatorCall.kt:37)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitExpression(IrElementTransformerVoid.kt:131)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitTypeOperator(IrElementTransformerVoid.kt:260)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitTypeOperator(IrElementTransformerVoid.kt:261)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitTypeOperator(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.expressions.IrTypeOperatorCall.accept(IrTypeOperatorCall.kt:30)
	at org.jetbrains.kotlin.ir.expressions.IrExpression.transform(IrExpression.kt:28)
	at org.jetbrains.kotlin.ir.expressions.IrReturn.transformChildren(IrReturn.kt:32)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitExpression(IrElementTransformerVoid.kt:131)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitReturn(IrElementTransformerVoid.kt:308)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitReturn(IrElementTransformerVoid.kt:309)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitReturn(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.expressions.IrReturn.accept(IrReturn.kt:25)
	at org.jetbrains.kotlin.ir.expressions.IrExpression.transform(IrExpression.kt:28)
	at org.jetbrains.kotlin.ir.expressions.IrExpression.transform(IrExpression.kt:21)
	at org.jetbrains.kotlin.ir.util.TransformKt.transformInPlace(transform.kt:35)
	at org.jetbrains.kotlin.ir.expressions.IrBlockBody.transformChildren(IrBlockBody.kt:31)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBody(IrElementTransformerVoid.kt:108)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlockBody(IrElementTransformerVoid.kt:117)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlockBody(IrElementTransformerVoid.kt:118)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlockBody(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.expressions.IrBlockBody.accept(IrBlockBody.kt:24)
	at org.jetbrains.kotlin.ir.expressions.IrBody.transform(IrBody.kt:21)
	at org.jetbrains.kotlin.ir.declarations.IrFunction.transformChildren(IrFunction.kt:61)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitDeclaration(IrElementTransformerVoid.kt:57)
	at org.jetbrains.kotlin.ir.backend.js.lower.JsCodeOutlineTransformer.visitDeclaration(JsCodeOutliningLowering.kt:117)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitFunction(IrElementTransformerVoid.kt:69)
	at org.jetbrains.kotlin.backend.common.IrElementTransformerVoidWithContext.visitFunctionNew(IrElementTransformerVoidWithContext.kt:130)
	at org.jetbrains.kotlin.backend.common.IrElementTransformerVoidWithContext.visitFunction(IrElementTransformerVoidWithContext.kt:83)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitSimpleFunction(IrElementTransformerVoid.kt:72)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitSimpleFunction(IrElementTransformerVoid.kt:73)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitSimpleFunction(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.declarations.IrSimpleFunction.accept(IrSimpleFunction.kt:36)
	at org.jetbrains.kotlin.ir.IrElementBase.transform(IrElementBase.kt:24)
	at org.jetbrains.kotlin.ir.declarations.IrProperty.transformChildren(IrProperty.kt:58)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitDeclaration(IrElementTransformerVoid.kt:57)
	at org.jetbrains.kotlin.ir.backend.js.lower.JsCodeOutlineTransformer.visitDeclaration(JsCodeOutliningLowering.kt:117)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitProperty(IrElementTransformerVoid.kt:78)
	at org.jetbrains.kotlin.backend.common.IrElementTransformerVoidWithContext.visitPropertyNew(IrElementTransformerVoidWithContext.kt:134)
	at org.jetbrains.kotlin.backend.common.IrElementTransformerVoidWithContext.visitProperty(IrElementTransformerVoidWithContext.kt:69)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitProperty(IrElementTransformerVoid.kt:79)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitProperty(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.declarations.IrProperty.accept(IrProperty.kt:48)
	at org.jetbrains.kotlin.ir.IrElementBase.transform(IrElementBase.kt:24)
	at org.jetbrains.kotlin.ir.util.TransformKt.transformInPlace(transform.kt:35)
	at org.jetbrains.kotlin.ir.declarations.IrClass.transformChildren(IrClass.kt:74)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitDeclaration(IrElementTransformerVoid.kt:57)
	at org.jetbrains.kotlin.ir.backend.js.lower.JsCodeOutlineTransformer.visitDeclaration(JsCodeOutliningLowering.kt:117)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitClass(IrElementTransformerVoid.kt:66)
	at org.jetbrains.kotlin.backend.common.IrElementTransformerVoidWithContext.visitClassNew(IrElementTransformerVoidWithContext.kt:126)
	at org.jetbrains.kotlin.backend.common.IrElementTransformerVoidWithContext.visitClass(IrElementTransformerVoidWithContext.kt:62)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitClass(IrElementTransformerVoid.kt:67)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitClass(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.declarations.IrClass.accept(IrClass.kt:64)
	at org.jetbrains.kotlin.ir.IrElementBase.transform(IrElementBase.kt:24)
	at org.jetbrains.kotlin.ir.util.TransformKt.transformInPlace(transform.kt:35)
	at org.jetbrains.kotlin.ir.expressions.IrContainerExpression.transformChildren(IrContainerExpression.kt:32)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitExpression(IrElementTransformerVoid.kt:131)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitContainerExpression(IrElementTransformerVoid.kt:166)
	at org.jetbrains.kotlin.ir.backend.js.lower.JsCodeOutlineTransformer.visitContainerExpression(JsCodeOutliningLowering.kt:113)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlock(IrElementTransformerVoid.kt:169)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlock(IrElementTransformerVoid.kt:170)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlock(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.expressions.IrBlock.accept(IrBlock.kt:22)
	at org.jetbrains.kotlin.ir.expressions.IrExpression.transform(IrExpression.kt:28)
	at org.jetbrains.kotlin.ir.expressions.IrMemberAccessExpression.transformChildren(IrMemberAccessExpression.kt:82)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitExpression(IrElementTransformerVoid.kt:131)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitDeclarationReference(IrElementTransformerVoid.kt:178)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitMemberAccess(IrElementTransformerVoid.kt:208)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitFunctionAccess(IrElementTransformerVoid.kt:211)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitConstructorCall(IrElementTransformerVoid.kt:217)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitConstructorCall(IrElementTransformerVoid.kt:218)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitConstructorCall(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.expressions.IrConstructorCall.accept(IrConstructorCall.kt:27)
	at org.jetbrains.kotlin.ir.expressions.IrExpression.transform(IrExpression.kt:28)
	at org.jetbrains.kotlin.ir.expressions.IrExpressionBody.transformChildren(IrExpressionBody.kt:35)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:346)
	at org.jetbrains.kotlin.ir.backend.js.lower.JsCodeOutliningLowering.lower(JsCodeOutliningLowering.kt:43)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor$visitBody$1.invoke(Lower.kt:190)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor$visitBody$1.invoke(Lower.kt:189)
	at org.jetbrains.kotlin.ir.backend.js.WholeWorldStageController.restrictTo(WholeWorldStageController.kt:29)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitBody(Lower.kt:189)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitBody(Lower.kt:168)
	at org.jetbrains.kotlin.ir.visitors.IrElementVisitor$DefaultImpls.visitExpressionBody(IrElementVisitor.kt:161)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitExpressionBody(Lower.kt:168)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitExpressionBody(Lower.kt:168)
	at org.jetbrains.kotlin.ir.expressions.IrExpressionBody.accept(IrExpressionBody.kt:25)
	at org.jetbrains.kotlin.ir.declarations.IrField.acceptChildren(IrField.kt:45)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitDeclaration(Lower.kt:177)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitDeclaration(Lower.kt:168)
	at org.jetbrains.kotlin.ir.visitors.IrElementVisitor$DefaultImpls.visitField(IrElementVisitor.kt:124)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitField(Lower.kt:168)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitField(Lower.kt:168)
	at org.jetbrains.kotlin.ir.declarations.IrField.accept(IrField.kt:42)
	at org.jetbrains.kotlin.ir.declarations.IrProperty.acceptChildren(IrProperty.kt:51)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitDeclaration(Lower.kt:177)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitDeclaration(Lower.kt:168)
	at org.jetbrains.kotlin.ir.visitors.IrElementVisitor$DefaultImpls.visitProperty(IrElementVisitor.kt:132)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitProperty(Lower.kt:168)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitProperty(Lower.kt:168)
	at org.jetbrains.kotlin.ir.declarations.IrProperty.accept(IrProperty.kt:48)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitClass(Lower.kt:183)
	at org.jetbrains.kotlin.backend.common.BodyLoweringVisitor.visitClass(Lower.kt:168)
	at org.jetbrains.kotlin.ir.declarations.IrClass.accept(IrClass.kt:64)
	at org.jetbrains.kotlin.backend.common.LowerKt.runOnFilePostfix(Lower.kt:147)
	at org.jetbrains.kotlin.backend.common.LowerKt.runOnFilePostfix$default(Lower.kt:140)
	at org.jetbrains.kotlin.backend.common.BodyLoweringPass.lower(Lower.kt:62)
	at org.jetbrains.kotlin.backend.common.LowerKt.lower(Lower.kt:75)
	at org.jetbrains.kotlin.ir.backend.js.JsLoweringPhasesKt$makeJsModulePhase$1.invoke(JsLoweringPhases.kt:36)
	at org.jetbrains.kotlin.ir.backend.js.JsLoweringPhasesKt$makeJsModulePhase$1.invoke(JsLoweringPhases.kt:35)
	at org.jetbrains.kotlin.ir.backend.js.JsLoweringPhasesKt$makeCustomJsModulePhase$1.invoke(JsLoweringPhases.kt:59)
	at org.jetbrains.kotlin.ir.backend.js.JsLoweringPhasesKt$makeCustomJsModulePhase$1.invoke(JsLoweringPhases.kt:51)
	at org.jetbrains.kotlin.backend.common.phaser.NamedCompilerPhase.invoke(CompilerPhase.kt:96)
	at org.jetbrains.kotlin.ir.backend.js.CompilerWithICKt.lowerPreservingTags(compilerWithIC.kt:76)
	at org.jetbrains.kotlin.ir.backend.js.JsIrCompilerWithIC.compile(compilerWithIC.kt:59)
	at org.jetbrains.kotlin.ir.backend.js.ic.CacheUpdater.compileDirtyFiles(CacheUpdater.kt:676)
	at org.jetbrains.kotlin.ir.backend.js.ic.CacheUpdater.actualizeCaches(CacheUpdater.kt:727)
	at org.jetbrains.kotlin.cli.js.K2JsIrCompiler.doExecute(K2JsIrCompiler.kt:300)
	at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:183)
	at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:72)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:101)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:47)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1645)
	at jdk.internal.reflect.GeneratedMethodAccessor99.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:587)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:705)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:704)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.jetbrains.kotlin.com.google.gwt.dev.js.parserExceptions.JsParserException: missing name after . operator at (51, 19)
	... 174 more
```

Unfortunately I don’t have the intermediate `.js` file that triggered this.

--- a/zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/CallChannel.kt
+++ b/zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/CallChannel.kt
@@ -17,6 +17,10 @@ package app.cash.zipline.internal.bridge
 
 import kotlin.js.JsName
 
+/**
+ * Be careful changing these. They're inlined manually in js() blocks to workaround this issue:
+ * https://github.com/cashapp/zipline/issues/900
+ */
 internal const val inboundChannelName = "app_cash_zipline_inboundChannel"
 internal const val outboundChannelName = "app_cash_zipline_outboundChannel"
 

--- a/zipline/src/jsMain/kotlin/app/cash/zipline/GlobalBridge.kt
+++ b/zipline/src/jsMain/kotlin/app/cash/zipline/GlobalBridge.kt
@@ -41,7 +41,7 @@ internal object GlobalBridge : JsPlatform, CallChannel {
     val globalBridge = this
     js(
       """
-      globalThis.$inboundChannelName = globalBridge;
+      globalThis.app_cash_zipline_inboundChannel = globalBridge;
 
       globalThis.setTimeout = function(handler, delay) {
         return globalBridge.setTimeout(handler, delay, arguments);

--- a/zipline/src/jsMain/kotlin/app/cash/zipline/ziplineJs.kt
+++ b/zipline/src/jsMain/kotlin/app/cash/zipline/ziplineJs.kt
@@ -49,7 +49,7 @@ actual class Zipline internal constructor(userSerializersModule: SerializersModu
       /** Lazily fetch the channel to call out. */
       @Suppress("UnsafeCastFromDynamic")
       private val jsOutboundChannel: CallChannel
-        get() = js("globalThis.$outboundChannelName")
+        get() = js("globalThis.app_cash_zipline_outboundChannel")
 
       override fun serviceNamesArray(): Array<String> {
         return jsOutboundChannel.serviceNamesArray()

